On build failed with compilation error:
```
> Task :streams:compileJava
--
more_vert | 2701 | 06:13:33 AM | /home/jenkins/workspace/Kafka_kafka-pr_PR-16985/streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java:481: error: incompatible types: inference variable K#1 has incompatible equality constraints V#2,K#2,K#3
  | 2702 | 06:13:33 AM | addSink(name, new StaticTopicNameExtractor<>(topic), keySerializer, valSerializer, partitioner, predecessorNames);
  | 2703 | 06:13:33 AM | ^
  | 2704 | 06:13:33 AM | where K#1,V#1,V#2,K#2,K#3 are type-variables:
  | 2705 | 06:13:33 AM | K#1 extends Object declared in method <K#1,V#1>addSink(String,TopicNameExtractor<K#1,V#1>,Serializer<K#1>,Serializer<V#1>,StreamPartitioner<? super K#1,? super V#1>,String...)
  | 2706 | 06:13:33 AM | V#1 extends Object declared in method <K#1,V#1>addSink(String,TopicNameExtractor<K#1,V#1>,Serializer<K#1>,Serializer<V#1>,StreamPartitioner<? super K#1,? super V#1>,String...)
  | 2707 | 06:13:33 AM | V#2 extends Object declared in method <K#2,V#2>addSink(String,String,Serializer<K#2>,Serializer<V#2>,StreamPartitioner<? super K#2,? super V#2>,String...)
  | 2708 | 06:13:33 AM | K#2 extends Object declared in method <K#2,V#2>addSink(String,String,Serializer<K#2>,Serializer<V#2>,StreamPartitioner<? super K#2,? super V#2>,String...)
  | 2709 | 06:13:33 AM | K#3 extends Object declared in class StaticTopicNameExtractor
  | 2720 | 06:13:33 AM | 1 error
  | 2721 | 06:13:33 AM |  
```

Not sure how the others could pass? -- There is also bunch of test failures; are they related?

--- a/streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java
+++ b/streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java
@@ -33,10 +33,8 @@ public class TaskId implements Comparable<TaskId> {
     public static final String NAMED_TOPOLOGY_DELIMITER = "__";
 
     /** The ID of the subtopology, aka topicGroupId. */
-    @Deprecated
     private final int topicGroupId;
     /** The ID of the partition. */
-    @Deprecated
     private final int partition;
 
     /** The namedTopology that this task belongs to, or null if it does not belong to one */

